admin/entities.py

```diff
@@ -103,6 +104,49 @@ class SubmissionAdminView(BaseAdminView):
 
 class OrganisationAdminView(BaseAdminView):
     _model = Organisation
+    instance = None
+    can_edit = True
+    fields_allowed_to_edit = ["external_reference_code"]
+    edit_template = "admin/edit_organisation.html"
+
+    column_list = [
+        Organisation.organisation_name,
+        Organisation.external_reference_code,
+        Organisation.organisation_type,
+    ]
+    column_labels = {
+        "external_reference_code": "External reference code",
+        "organisation_type": "Organisation type",
+    }
+    column_sortable_list = [
+        Organisation.organisation_name,
+    ]
+    column_searchable_list = [Organisation.organisation_name, Organisation.external_reference_code]
+    column_default_sort = [("organisation_name", False)]
+    form_excluded_columns = [
+        "programmes",
+        "organisation_name",
+        "organisation_type",
+    ]
+
+    @expose("/edit/", methods=("GET", "POST"))
+    def edit_view(self):
+        if request.method == "GET":
+            id = request.args.get("id")
+            if id:
+                self.instance = self.session.query(self.model).get(id)
+                if self.instance:
+                    self._template_args["organisation_name"] = self.instance.organisation_name
+
+        return super(OrganisationAdminView, self).edit_view()
+
+    def on_model_change(self, form, model, is_created):
+        if not is_created:
+            for field in form.data:
+                if field not in self.fields_allowed_to_edit and hasattr(model, field):
+                    setattr(model, field, getattr(self.instance, field))
+
+        return super(OrganisationAdminView, self).on_model_change(form, model, is_created)
```

Comment:
This is important behaviour to get correct - could we add some tests around this?

Reply:
Added a test.